### PR TITLE
`tan()` - New CSS functional notation

### PR DESCRIPTION
**Pull request by ramiy**
#### Summary
Document the CSS `tan()` functional notation.

#### Motivation
[Safari/webkit already supports](https://webkit.org/blog/12445/new-webkit-features-in-safari-15-4/) this CSS function. [Chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=1190444) will follow soon.

#### Supporting details
https://drafts.csswg.org/css-values/#trig-funcs

#### Metadata
- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by ramiy** [comment]
@estelle Do you have a suggestion for `tan()` example?

**Comment by teoli2003** [comment]
There is a parallelogram example using `tan()` there: https://github.com/w3c/csswg-drafts/issues/2331#issuecomment-366469494

It may be the basis for an example here.

**Comment by ramiy** [comment]
@teoli2003 please review or send feedback.

File: files/en-us/web/css/tan/index.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  - Math
+  - Reference
+  - Web
+  - tan
+browser-compat: css.types.tan
+spec-urls: https://drafts.csswg.org/css-values/#trig-funcs
+---
+{{CSSRef}}
+
+The **`tan()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) is a trigonometric function that returns the tangent of a number, which is a value between `−∞` and `∞`. The function contains a single calculation that must resolve to either a {{cssxref("&lt;number&gt;")}} or an {{cssxref("&lt;angle&gt;")}} by interpreting the result of the argument as radians. That is, `tan(0deg)` will return `0`, `tan(45deg)` will return `0.9999999999999999` and `tan(90deg)` will return `16331239353195370`.
```
Comment (teoli2003) [inline files/en-us/web/css/tan/index.md:17]: >  and `tan(90deg)` will return `16331239353195370`

This is not what the spec says:
> In tan(A), if A is one of the asymptote values (such as 90deg, 270deg, etc), the result must be +∞ for 90deg and all values a multiple of 360deg from that (such as -270deg or 450deg), and −∞ for -90deg and all values a multiple of 360deg from that (such as -450deg or 270deg).

How is +∞ interpreted as a `<number>`?

Comment (ramiy) [reply files/en-us/web/css/tan/index.md:17]: I can remove this for now.

Later on, we can add this to the "Syntax" section, talking about the argument (https://www.w3.org/TR/css-values-4/#trig-infinities).

Comment (ramiy) [reply files/en-us/web/css/tan/index.md]: @teoli2003

**Comment by ramiy** [comment]
Hi @estelle,
You suggested a very good example for css `cos()` function.
Can you suggest a better example for css `tan()` function?

File: files/en-us/web/css/tan/index.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  --angle: calc( sin( var(--h) / var(--w) ) );
+  content: '';
+  position: absolute;
+  width: calc( 100% - 100% * var(--h) / var(--w) * tan( var(--angle) ) );
+  height: 100%;
+  transform-origin: 0 100%;
+  transform: skew( calc( 0 - var(--angle) ) );
+  background-color: red;
+}
+```
```
Comment (teoli2003) [inline files/en-us/web/css/tan/index.md:69]: The live example on `cos()` is nice. Can we have one here too?

File: files/en-us/web/css/tan/index.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+- `angle`
+  - : A {{cssxref("&lt;number&gt;")}} or an {{cssxref("&lt;angle&gt;")}}. When specifying unitless numbers they are interpreted as a number of radians, representing an {{cssxref("&lt;angle&gt;")}}. When specifying `infinity`, `-infinity`, or `NaN`, the result is `NaN`. When specifying one of the asymptote values (such as `90deg`, `270deg`, etc), the result must be `infinity` for `90deg` and all values a multiple of `360deg` from that (such as `-270deg` or `450deg`), and `−infinity` for `-90deg` and all values a multiple of `360deg` from that (such as `-450deg` or `270deg`).
+
+### Formal syntax
+
+{{CSSSyntax}}
+
+## Examples
+
+### Drawing Parallelogram
```
Comment (teoli2003) [inline files/en-us/web/css/tan/index.md:49]: And we should replace this title with a text as there is only one example.

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Looks good, can we make the example a live example? This would be nice!

File: files/en-us/web/css/tan/index.md (hunk 2 above)
Comment (ramiy) [reply files/en-us/web/css/tan/index.md:69]: Yes we can, but I tested the code in Firefox (with the css trig flag enabled), it doesn't work...

**Comment by ramiy** [comment]
@teoli2003 live example added.

**Review by teoli2003 — APPROVED** [review]

**Review by bsmth — APPROVED** [review]
hi @ramiy, thanks a lot for the additions 🙌🏻

**Pull request merged**